index.d.ts

```diff
@@ -0,0 +1,39 @@
+declare module 'envvar' {
+  /**
+   * Retrieves a given environment variable as a string.
+   *
+   * @throws if the environment variable is not set and a default is not
+   *         specified.
+   * @throws if the value of the environment variable is not a string.
+   */
+  export function string(name: string, defaultValue?: string): string;
+
+  /**
+   * Retrieves a given environment variable as a number.
+   *
+   * @throws if the environment variable is not set and a default is not
+   *         specified.
+   * @throws if the value of the environment variable is not a number.
+   */
+  export function number(name: string, defaultValue?: number): number;
+
+  /**
+   * Retrieves a given environment variable as a boolean.
+   *
+   * @throws if the environment variable is not set and a default is not
+   *         specified.
+   * @throws if the value of the environment variable is not a boolean.
+   */
+  export function boolean(name: string, defaultValue?: boolean): boolean;
+
+  /**
+   * Retrieves a given environment variable, validating that its value is
+   * contained within the specified enum.
+   *
+   * @throws if the environment variable is not set and a default is not
+   *         specified.
+   * @throws if the value of the environment variable is not a string
+   *         that is one of the allowed values.
```

Comment:
How about this phrasing?

_Throws if the value of the environment variable is not one of the permissible strings._